examples/lite/lite.py

```diff
@@ -0,0 +1,230 @@
+import tensorflow as tf
+
+
+import jax
+from reinforced_lib.agents.mab import ThompsonSampling
+from functools import wraps
+import jax.numpy as jnp
+import jax.tree_util as tree
+
+def flatten_args(tree_args_fun, treedef):
+    @wraps(tree_args_fun)
+    def flat_args_fun(*leaves):
+        tree_args = tree.tree_unflatten(treedef, leaves)
+        tree_ret = tree_args_fun(*tree_args)
+        return tree.tree_leaves(tree_ret)
+
+    return flat_args_fun
+
+def make_converter(f,example):
+    """
+
+    Parameters
+    ----------
+    f a function
+    example arguments in tree format
+    """
+    leaves,treedef = tree.tree_flatten(example)
+
+    flat_fun = flatten_args(f,treedef)
+
+    inputs = [[ (f'arg{i}',l) for i,l in enumerate(leaves)]]
+    converter = tf.lite.TFLiteConverter.experimental_from_jax([flat_fun],inputs)
+    return converter
+
+def main():
+    k = jax.random.PRNGKey(42)
+    ts = ThompsonSampling(16)
+
+    state = ts.init(k)
+    new_state = ts.update(state,k,1,2,1,0.1)
```

Comment:
@m-wojnar I am not sure how to handle keys in tf.lite. Maybe we could store the key in agent state as merge split into update function?

Reply:
This would require a few changes to the structure of the library itself and updating all agents. There is also a big problem with the `sample` method if it uses a key. We intentionally blocked the possibility of updating the agent state in the `sample` method, so after moving the key to the state, we are not able to update this key after split (thus, if we want to call `sample` several times without calling `update,` the same action will be returned each time).

It would be ideal to handle the key in TF Lite. However, if that's not possible, we can try to figure out another way to store the key in a state and then handle it in the `sample` and `update` methods.

Reply:
Ok, so we need to export `jax.random.split` and this solves the problem.

Reply:
Or we can create a wrapper for the agent, whose task will only be to launch the methods of the appropriate agent and manage the key. In addition, this class could contain a method to export to TF Lite and it does not have to inherit from `BaseAgent`. Sample pseudocode:

```
@dataclass
class TfLiteState:
    agent_state: Any
    key: PRNGKey

class TfLiteAgent:
    def __init__(self, agent: BaseAgent) -> None:
        self.init = partial(self.init, agent=agent)
        self.sample = partial(self.sample, agent=agent)
        self.update = partial(self.update, agent=agent)

    def init(self, agent: BaseAgent, *args, **kwargs) -> TfLiteState:
        return TfLiteState(
            agent_state=agent.init(*args, **kwargs),
            key=jax.random.PRNGKey(42)
        )

    def sample(self, agent: BaseAgent, state: TfLiteState, *args, **kwargs) -> Tuple[Any, TfLiteState]:
        sample_key, key = jax.random.split(state.key)
        action = agent.sample(state.agent_state, sample_key, *args, **kwargs)
        return action, TfLiteState(agent_state=state, key=key)

    def update(self, agent: BaseAgent, state: TfLiteState, *args, **kwargs) -> Tuple[Any, TfLiteState]:
        update_key, key = jax.random.split(state.key)
        new_state = agent.update(state.agent_state, update_key, *args, **kwargs)
        return action, TfLiteState(agent_state=new_state, key=key)
```

Reply:
Makes sense, this wrapper could also translate API, to flat list using `@flatten_args`